urllib3/connection.py

```diff
@@ -162,6 +162,27 @@ def connect(self):
         conn = self._new_conn()
         self._prepare_conn(conn)
 
+    def request_chunked(self, method, url, body=None, headers={}):
+        """ 
+        Alternative to the common request method, which sends the 
+        body with chunked encoding and not as one block 
+        """
+        header_names = set(k.lower() for k in headers)
+        self.putrequest(method, url, skip_accept_encoding=True)
+        for header, value in headers.items():
+            self.putheader(header, value)
+        if 'transfer-encoding' not in headers:
+            self.putheader('Transfer-Encoding', 'chunked')
```

Comment:
Just saw another bug here. It should check for in header_names, not headers.

Is it allowed to define several encodings in the headers? Like chunked gzip? I don't think so. Maybe it should throw an error, if request_chunked is run with another encoding set?

Side note: The original request also sets a transfer-encoding automatically (identity), if none is set.